### PR TITLE
add a kickstart problem to LCA section

### PR DESCRIPTION
**Pull request by xyzyzl**
problem is pretty informative on how LCA can be modified to store additional information

_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

**Comment by xyzyzl** [comment]
i believe current issue is backend

File: solutions/ks-dependent.mdx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+using ll = long long;
+const int MOD = 1e9+7;
+
+// due to Benjamin Qi
```
Comment (nwatx) [inline solutions/ks-dependent.mdx:23]: ?

File: solutions/ks-dependent.mdx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	int inv(int b, int mod = MOD) { return po(b, mod - 2, mod); }
+
+	int invGeneral(int a, int b)
+	{ // 0 < a < b, gcd(a,b) = 1
+		if (a == 0)
+			return b == 1 ? 0 : -1;
+		int x = invGeneral(b % a, a);
+		return x == -1 ? -1 : ((1 - (ll)b * x) / a + b) % b;
+	}
+} // namespace modOp
```
Comment (nwatx) [inline solutions/ks-dependent.mdx:45]: should use ModInt instead for readability

File: solutions/ks-dependent.mdx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+using namespace modOp;
+
+const int D = inv(1000000);
```
Comment (nwatx) [inline solutions/ks-dependent.mdx:49]: comment/explain what this does

**Review by nwatx — CHANGES_REQUESTED** [review]

File: solutions/ks-dependent.mdx (hunk 1 above)
Comment (xyzyzl) [reply solutions/ks-dependent.mdx:23]: yeah my bad that was from back when u could "credit" code snippets in usaco

File: solutions/ks-dependent.mdx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+			p[x] = ad(mul(pa[x][0], p[par[x][0]]), mul(pb[x][0], sub(1, p[par[x][0]])));
+			dfs(x);
+		}
+	};
+	dfs(0); // initializes the binary jump parents, as well as the fixed probabilities of each node
+
+	function<int(int,int)> lca = [&](int u, int v)
+	{
+		if(dep[u] < dep[v]) swap(u,v);
+		int ua=1, ub=0, va=1, vb=0;
```
Comment (nwatx) [inline solutions/ks-dependent.mdx:91]: operator spacing

**Comment by bqi343** [comment]
I'll let @nwatx handle this.

**Review by nwatx — APPROVED** [review]

**Comment by nwatx** [comment]
@SansPapyrus683 can you review this

**Comment by SansPapyrus683** [comment]
want me to go all out or...?

**Comment by nwatx** [comment]
> want me to go all out or...?

sure?

**Comment by SansPapyrus683** [comment]
oh god this code is cursed beyond belief

File: solutions/ks-dependent.mdx [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+	mint& operator++() { return *this += 1; }
+	mint& operator--() { return *this -= 1; }
+	friend mint operator+(mint a, const mint& b) { return a += b; }
+	friend mint operator-(mint a, const mint& b) { return a -= b; }
+	friend mint operator*(mint a, const mint& b) { return a *= b; }
+	friend mint operator/(mint a, const mint& b) { return a /= b; }
+};
+
+using mi = mint<MOD,5>; // 5 is primitive root for both common mods
+using vmi = vector<mi>;
```
[the remark this answers is not in the document]
Comment (SansPapyrus683) [reply solutions/ks-dependent.mdx:64]: no don't do this

File: solutions/ks-dependent.mdx [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+		mi x = lca(u,v);
+		cout << x.v << ' ';
+	}
+
+	cout << endl;
+}
+
+int main() {
+	int t; cin >> t;
+	for(int i = 1; i <= t; i++) {
```
Comment (SansPapyrus683) [inline solutions/ks-dependent.mdx:171]: spaces for control flow statements

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
dear lord

**Comment by SansPapyrus683** [comment]
neo i know most of my suggestions will be rejected, you can merge w/o me

**Comment by SansPapyrus683** [comment]
lol love how you just gave up

**Pull request merged**